services/wallet/router/router_v2.go

```diff
@@ -417,26 +418,41 @@ func validateInputData(input *RouteInputParams) error {
 	return nil
 }
 
+func (r *Router) SuggestedRoutesV2Async(input *RouteInputParams) {
+	r.scheduler.Enqueue(async.TaskType{
```

Comment:
@saledjenic, I'm just curious more than anything, so no suggestions for change.

1. I'm assuming we are going in the async path because calculating the routes can take a while, correct? Does the queue preemptively kill tasks that take too long? Is it "guaranteed" that getting suggested routes will finish eventually and within a reasonable amount of time?
2. I don't know if it would be useful, but in case getting suggested routes takes too long, clients may find useful to receive a token/identifier from this endpoint that they can later use at their own discretion to cancel enqueued tasks (for example to dispatch again `SuggestedRoutesV2Async` if they detect connection issues or based on some other state outside status-go).

Reply:
> I'm assuming we are going in the async path because calculating the routes can take a while, correct?

Correct.

> Does the queue preemptively kill tasks that take too long?

You can see that the set policy is `ReplacementPolicyCancelOld`, which means the current task will be canceled and replaced with a new one, you can see the implementation details here https://github.com/status-im/status-go/blob/develop/services/wallet/async/scheduler.go#L67-L133

> Is it "guaranteed" that getting suggested routes will finish eventually and within a reasonable amount of time?

The router algorithm does many external calls, and for each of them we have a timeout set, so if something takes long will be canceled after the timeout.

> I don't know if it would be useful, but in case getting suggested routes takes too long, clients may find it useful to receive a token/identifier from this endpoint that they can later use at their own discretion to cancel enqueued tasks

The UI is designed in a way that the user cannot ask for the route and then close that popup and switch to some other view. But you're probably referring to if the user wants to close that popup we should explicitly close the running task, correct? That's not how we're doing it now, we're closing the popup, task remains till the end of its execution, we just don't respond to the received signal in that case (that's how it is/was). 

But thinking, since we're doing some changes now, maybe it's a good time to introduce that and will be on the client how will use it, whether to leave task running (ignoring the signal) or explicitly cancel it.

> for example to dispatch again SuggestedRoutesV2Async if they detect connection issues

If there are connection issues the route can't be found, cause the first external call after disconnection will return an error.

Reply:
Thank you @saledjenic for the insightful answers.

> maybe it's a good time to introduce that and will be on the client how will use it, whether to leave task running (ignoring the signal) or explicitly cancel it

At least on mobile, when a signal arrives and we ignore it, it still takes resources because it has to go through a queue processed by re-frame (a lib managing all sorts of events in the app) which competes for attention from other UI events in this single-threaded environment, among other details. Nothing too important, but whenever a signal is not published it helps a bit with performance.

I know my comment about making this (potentially) long-running signal cancellable  is leaning on the side of over-engineering, so please ignore it if it feels that way.

Reply:
It's already added, and there is now the `StopSuggestedRoutesV2AsyncCalcualtion` endpoint you can call from the mobile app when the user is about to close the sending page.